api/base/filters.py

```diff
@@ -541,10 +541,12 @@ def param_queryset(self, query_params, default_queryset):
         if filters:
             for key, field_names in filters.iteritems():
                 for field_name, data in field_names.iteritems():
-                    if isinstance(queryset, list):
-                        queryset = self.get_filtered_queryset(field_name, data, queryset)
-                    else:
-                        queryset = self.filter_by_field(queryset, field_name=field_name, operation=data)
+                    operations = [item for item in data] if isinstance(data, list) else [data]
```

Comment:
Chances were also good that I was about to learn something interesting about a quirk in our system, but it's nice to know I can make a contribution every now and then. ;)